src/components/Rating/Rating.tsx

```diff
@@ -0,0 +1,59 @@
+import StarIcon from '@/public/svgs/star.svg';
+import classNames from 'classnames/bind';
+import { useState } from 'react';
+import styles from './Rating.module.scss';
+
+interface StarProps {
+  checked: boolean;
+  onClick?: () => void;
+  onMouseOver?: () => void;
+  onMouseOut?: () => void;
+  isEditable?: boolean;
+}
+
+interface RatingProps {
+  rating: number;
+  onRatingChange?: (rating: number) => void;
+  isEditable?: boolean;
+}
+
+const cn = classNames.bind(styles);
+
+function Star({ checked, onClick, onMouseOver, onMouseOut, isEditable }: StarProps) {
+  return (
+    <StarIcon
+      className={cn(checked ? 'checked' : 'not-checked', isEditable && 'edit')}
+      onClick={onClick}
+      onMouseOver={onMouseOver}
+      onMouseOut={onMouseOut}
```

Comment:
```suggestion
      onClick={() => isEditable && onClick()}
      onMouseOver={() => isEditable && onMouseOver()}
      onMouseOut={() => isEditable && onMouseOut()}
```
제대로 작동되는지는 한번 확인해봐아겠지만, 이런 식으로 넘겨 주시면, 사용처(Rating)에서는 삼항 연산자 필요 없이 다음과 같이 사용 가능하지 않을까요?
```typescript
return (
  <div>
    {STARS.map((star) => (
      <Star
        key={star}
        checked={star <= (hoverRating || rating)}
        onClick={() => handleRating(star)}
        onMouseOver={() => star > rating && setHoverRating(star)}
        onMouseOut={() => setHoverRating(0)}
        isEditable={isEditable}
      />))}
  </div>
);
```

Reply:
오홍 넵 분리하면 더 보기 편할 것 같았는데 알려주신 것도 좋은 것 같아요!  반영했습니다!